api/oneclick.py

```diff
@@ -0,0 +1,214 @@
+import logging
+from nose.tools import set_trace
+#from lxml import etree
+#from urlparse import urljoin
+#from urllib import urlencode
+#import datetime
+import requests
+
+#from core.util.xmlparser import XMLParser
+from authenticator import BasicAuthAuthenticator
+#from config import Configuration
+#import os
+#import re
+
+from circulation import (
+    #LoanInfo,
+    #FulfillmentInfo,
+    #HoldInfo,
+    BaseCirculationAPI
+)
+from circulation_exceptions import *
+
+from core.oneclick import (
+    OneClickAPI as BaseOneClickAPI,
+    MockOneClickAPI as BaseMockOneClickAPI,
+    #Axis360Parser,
+    #BibliographicParser,
+    OneClickBibliographicCoverageProvider
+)
+
+from core.model import (
+#    get_one,
+#    get_one_or_create,
+    Patron,
+)
+
+
+
+class OneClickAPI(BaseOneClickAPI, BasicAuthAuthenticator, BaseCirculationAPI):
+
+    NAME = "OneClick"
+    
+    '''
+    RECORD_NUMBER_FIELD = 'RECORD #[p81]'
+    PATRON_TYPE_FIELD = 'P TYPE[p47]'
+    EXPIRATION_FIELD = 'EXP DATE[p43]'
+    BARCODE_FIELD = 'P BARCODE[pb]'
+    USERNAME_FIELD = 'ALT ID[pu]'
+    FINES_FIELD = 'MONEY OWED[p96]'
+    '''
+    EXPIRATION_DATE_FORMAT = '%m-%d-%y'
+
+    #MULTIVALUE_FIELDS = set(['NOTE[px]', BARCODE_FIELD])
+
+    #REPORTED_LOST = re.compile("^CARD([0-9]{14})REPORTEDLOST")
+
+    # How long we should go before syncing our internal Patron record
+    # with Millenium.
+    #MAX_STALE_TIME = datetime.timedelta(hours=12)
+
+    log = logging.getLogger("OneClick Patron API")
+
+
+    def __init__(self, *args, **kwargs):
+        super(OneClickAPI, self).__init__(*args, **kwargs)
+        self.bibliographic_coverage_provider = (
+            OneClickBibliographicCoverageProvider(
+                self._db, oneclick_api=self
+            )
+        )
+
+
+    def checkout_item(self, patron_id, item_id, return_item=False):
+        """
+        :param patron_id OneClick internal id
+        :param item_id isbn
+        :return information on the transaction or error status and message
+        """
+        url = "%s/libraries/%s/patrons/%s/checkouts/%s" % (self.base_url, str(self.library_id), patron_id, item_id)
+
+        method = "post"
+        if return_item:
+            method = "delete"
```

Comment:
This makes me think the method shouldn't be called `checkout_item`--that `checkout_item` and `return_item` should be wrappers around a method with a more generic name.